### PR TITLE
Super linter and readme

### PR DESCRIPTION
**Pull request by admiralAwkbar**
# Update README and validate

- Using Super-Linter to validate all code
- Adding documentation on `README` to help explain the flow and usage of the tool

## Additional Info

- [ ] Did you write/update appropriate tests
- [ ] Release notes updated (if appropriate)
- [ ] Appropriate logging output
- [ ] Issue linked

### CODE REVIEW

**Comment by admiralAwkbar** [comment]
@dylan-smith @dmckinstry Added some nice notes on how to use the tool.
Also did a little cleanup with the super-linter to make the code happier:)

Up for any ideas or collab

thanks all!

**Comment by dylan-smith** [comment]
If we're going to use super-linter (which I think is a great idea), can you have it run in our pre-existing PR CI so we can enforce it on PR's.

**Comment by admiralAwkbar** [comment]
Hey @dylan-smith We can add it as a `branch protection rule` if that's what you're asking.
Sorry, my brain is fried this week...

**Comment by admiralAwkbar** [comment]
it has a lot of configs to tune, I would assume maybe turning of `jscpd` as its a pain lol

**Comment by dylan-smith** [comment]
it looks like you have it configured to run on pushes to feature branches, could we change it to run on pr trigger? 

originally I was thinking we should incorporate the check into our existing CI workflow; but whether it's a separate job in the same yaml, or a separate workflow entirely I guess doesn't matter too much.

**Comment by admiralAwkbar** [comment]
@dylan-smith Hey that makes more sense :)

Yeah, no issues there. Feel free to make it kick however you see fit for the project.

Just wanted to use it at least 1 time to get a good idea of the project and whats hiding :)

**Comment by admiralAwkbar** [comment]
@dylan-smith That would be a good conversation with the main devs of this project to see when they want the `super-linter` to really give them the business :)

**Comment by admiralAwkbar** [comment]
@dylan-smith so this is odd im not sure what its mad about
I tried to update line endings but maybe I'm missing something?
Will play with more later

**Comment by dylan-smith** [comment]
It might be a good idea to split this into 2x PR's for superlinter and readme updates

File: README.md [hunk 1]
```diff
@@ -43,6 +43,63 @@ Commands:
 
 To generate a script, you'll need to set an `ADO_PAT` as an environment variable. Performing any of the commands that touch GitHub will need the `GH_PAT` environment variable.
 
+
+## Prerequisites
+
+- You will need at least `admin` access to the repository on **Azure DevOps**
```
[the remark this answers is not in the document]
Comment (admiralAwkbar) [reply README.md:49]: thank you, updated

File: README.md [hunk 2]
```diff
@@ -43,6 +43,63 @@ Commands:
...
+  - **Note:** Higher permissions may be needed as the tooling progresses
+- You will need the ability to `create repositories` in your Organization on **github.com**
```
[the remark this answers is not in the document]
Comment (admiralAwkbar) [reply README.md:51]: updated!

File: README.md [hunk 3]
```diff
@@ -43,6 +43,63 @@ Commands:
...
+  - **Note:** You will need `Org:Admin`, `use:all`, and `repo:all` to use the command-line **GH-GEI** tool that makes a **High Fidelity Migration**
+  - **Note:** Your `Personal Access Token` will also need to be authorized for usage against the Organization
+
+## Limitations
```
[the remark this answers is not in the document]
Comment (admiralAwkbar) [reply README.md:55]: yeah, this should outline that process. Which of the objects isn't transferred?

**Comment by admiralAwkbar** [comment]
@dylan-smith There were Windows-style line endings on that YAML file lol... a tough one to see

**Comment by dylan-smith** [comment]
@roferg already split the readme changes out and merged them in another PR where he expanded upon them.  I plan to do the same with the super linter changes, stay tuned.

**Comment by dylan-smith** [comment]
I'm going to close this PR, the readme changes were made in another PR, and I feel like we have enough static analysis for now. I'm sure super-linter would be an improvement, but it's just not a priority for our team to focus on this right now.

**Pull request closed without merging**